### PR TITLE
[Build] bundle OpenSearch Dashboards

### PR DESCRIPTION
**Pull request by kavilla**
### Description
Produce an OpenSearch Dashboards bundle with plugins provided by a manifest.

This requires the manifest to be in order with plugins first and then Dashboards
being last. The plugins will need to pulled down and placed in the plugins
folder. Then we need to bootstrap to generate the modules so that we can actually
install the plugins to the release build.

What is missing or not verified
* Environment setup, this requires nvm and yarn to be already set up.
We will need to set that up.
* Dropping the custom config into the bundle
* Signing of the artifact
* Tests

Signed-off-by: Kawika Avilla <kavilla414@gmail.com>

### Issues Resolved
Partial https://github.com/opensearch-project/opensearch-build/issues/158

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kavilla** [comment]
Previous PR: https://github.com/opensearch-project/opensearch-build/pull/435

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Good. I can take the refactoring of the dashboards `if` on top of this.

File: bundle-workflow/scripts/default/build.sh [hunk 1]
```diff
@@ -55,15 +59,32 @@ if [ -z "$VERSION" ]; then
...
+else
+    ./gradlew assemble --no-daemon --refresh-dependencies -DskipTests=true -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT
+
+    zipPath=$(find . -path \*build/distributions/*.zip)
+    distributions="$(dirname "${zipPath}")"
+
+    echo "COPY ${distributions}/*.zip"
+    mkdir -p $OUTPUT/plugins
+    cp ${distributions}/*.zip ./$OUTPUT/plugins
+fi
```
Comment (saratvemulapalli) [inline bundle-workflow/scripts/default/build.sh:88]: We should test these scripts as part of our CI workflows.
A follow up for both OpenSearch and Dashboards.

**Review by saratvemulapalli — APPROVED** [review]
This is nice, love to see Dashboards coming along quickly.

File: bundle-workflow/scripts/default/build.sh (hunk 1 above)
Comment (saratvemulapalli) [reply bundle-workflow/scripts/default/build.sh:88]: Opened up an issue: https://github.com/opensearch-project/opensearch-build/issues/591

**Pull request merged**